File: Cpp/1365_How_Many_Numbers_Are_Smaller_Than_the_Current_Number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> result(nums.size());
        map<int, int> count;
        for(auto& num : nums){
            ++count[num];
        }
        
        int sum = 0;
        for(auto& it : count){
            int temp = it.second;
            it.second = sum;
            sum += temp;
        }
        
        for(int i=0; i<nums.size(); ++i){
            result[i] = count[nums[i]];
        }
        
        return result;
    }
};

File: Kotlin/1365_How_Many_Numbers_Are_Smaller_Than_the_Current_Number.kt
class Solution {
    fun smallerNumbersThanCurrent(nums: IntArray): IntArray {
        val result = IntArray(nums.size)
        val count = IntArray(101){0}
        
        for(num in nums) {
            count[num] += 1
        }
        
        var sum = 0
        for(i in count.indices) {
            if(count[i] == 0) continue
            
            val temp = count[i]
            count[i] = sum
            sum += temp
        }
        
        for(i in nums.indices) {
            result[i] = count[nums[i]]
        }
        
        return result
    }
}